app/models/user.rb

```diff
@@ -86,6 +86,14 @@ def active_profile
     profiles.find(&:active?)
   end
 
+  # This user's most recently activated profile that has also been deactivated
+  # due to a password reset, or nil if there is no such profile
+  # @return [Profile,nil]
```

Comment:
I really like YARD-ing code, especially for types, since sometimes it's not always clear what the return type of code is in Ruby (and especially when it's a value or nil like here)